sdjournal/journal_test.go

```diff
@@ -343,3 +344,54 @@ func TestJournalReaderSmallReadBuffer(t *testing.T) {
 		t.Fatalf("Got unexpected message %s", got[1])
 	}
 }
+
+func TestJournalGetUniqueValues(t *testing.T) {
+	j, err := NewJournal()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer j.Close()
+
+	uniqueString := generateRandomField(20)
+	testEntries := []string{"A", "B", "C", "D"}
+	for _, v := range testEntries {
+		err := journal.Send("TEST: "+uniqueString, journal.PriInfo, map[string]string{uniqueString: v})
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Millisecond * 250)
```

Comment:
My previous suggestion was actually to have something similar to [waitAndNext]( https://github.com/coreos/go-systemd/blob/master/sdjournal/journal_test.go#L110) here, so that you can be confident that your latest entry is available when reading. Does it sound reasonable? Have you experienced races here before introducing the sleep?